docs/architecture.md

```diff
@@ -2,7 +2,38 @@
 
 This doc covers the architectural design of Kubeless and directory structure of the repository.
 
-## Architecture
+## Concepts
+
+Kubeless is built around below core concepts:
+
+- Functions
+- Triggers
+- Runtime
+
+### Functions
+
+Function is representation of the code to be executed, with specification and metadata like runtime dependencies, build instructions etc. A function has a independent life-cycle and allows below method to control it:
+
+* Deploy - deploy function as function instances. This step may involve building the function image or re-use pre-generated image and deploying it on the cluster.
+* Execute - invoke a function directly i.e) not through any event source
+* Get - Returns the function metadata and spec 
+* Update - Modify the function specification and its meta-data
+* Delete - Deletes a function, and clean up any resource provisioned for the function from the cluster
+* List - Show the list of functions and their metadata
+* Logs - Return the logs generated by a function
+
+### Triggers
+
+Trigger represents an event source and associated functions with it. When an event occurs in the event source, Kubeless shall ensure the associated functions are invoked. Kubeless supports 1:1 or 1:n association between the trigger and function dependeing on the event source type. Trigger is decoupled from the life-cycle of associated functions and can be independently operated with below methods:
```

Comment:
The same, I suggest:

A _Trigger_ represents an event source the functions associated to it. When an event occurs in the event source, Kubeless will ensure that the associated function are invoked **at most once**. A Trigger can be associated to a single function or to several ones depending on the event source type. They are decoupled from the life-cycle of functions and can be independently operated with the following methods: